File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        dict1 = { }
        for c1 in s1:
            if c1 not in dict1:
                dict1[c1] = 1
            else:
                dict1[c1]+=1
        dict2={ }
        for c2 in s2:
            if c2 not in dict2:
                dict2[c2] = 1
            else:
                dict2[c2]+=1
        ans = dict1.items() == dict2.items()
        if ans == False:
            return False
        
        
        count = 0
        for i in range(0,len(s1)):
            if s1[i]!=s2[i]:
                count+=1
        if count ==0 or count==2:
            return True
        return False

File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        HashMap<Character,Integer> hm = new HashMap<>();
        for(char w : s1.toCharArray()){
            hm.put(w,hm.getOrDefault(w,0)+1);
        }

        HashMap<Character,Integer> hm2 = new HashMap<>();
        for(char w2 : s2.toCharArray()){
            hm2.put(w2,hm2.getOrDefault(w2,0)+1);
        }
        boolean ans = hm.entrySet().equals(hm2.entrySet());
        if(ans == false){
            return false;
        }
        
        int count = 0;
        for(int i=0;i<s1.length();i++){
            if(s1.charAt(i)!=s2.charAt(i)){
                count+=1;
            }
        }
        if(count==0 || count==2){
            return true;
        }
        return  false;
    }
}